### PR TITLE
Refactor Join Page and Status Page JS

### PR DESCRIPTION
**Pull request by daltonfury42**

### CODE REVIEW

File: simplq/src/components/pages/Join/index.jsx [hunk 1]
```diff
@@ -14,28 +15,33 @@ import InputField from '../../common/InputField';
 
 export function JoinQueueWithDetails(props) {
   const queueId = props.match.params.queueId;
-  const queueName = useSelector((state) => state.appReducer.queueName);
+  const [queueStatusResponse, setQueueStatusResponse] = useState();
+  useEffect(
+    () => QueueService.getStatus(queueId).then(setQueueStatusResponse).catch(handleApiErrors),
+    [queueId]
```
Comment (maaverik) [inline simplq/src/components/pages/Join/index.jsx:21]: Does setQueueStatusResponse get called with the status argument here automatically? I didn't realize this syntax worked.

File: simplq/src/components/pages/Join/index.jsx [hunk 2]
```diff
@@ -14,28 +15,33 @@ import InputField from '../../common/InputField';
 
 export function JoinQueueWithDetails(props) {
   const queueId = props.match.params.queueId;
```
Comment (maaverik) [inline simplq/src/components/pages/Join/index.jsx:17]: I'll split up these components into different files and name them better. I intended to do it later but forgot.

Comment (daltonfury42) [reply simplq/src/components/pages/Join/index.jsx:17]: No lets not. As per https://github.com/SimplQ/simplQ-frontend/issues/121 we would ditch that intermediate page...

And always when you get such ideas during development, raise an issue and move on. It's pretty effective. You get bonus points if you leave a TODO comment in the code as well with a link to the issue.

File: simplq/src/components/pages/Join/index.jsx (hunk 1 above)
Comment (daltonfury42) [reply simplq/src/components/pages/Join/index.jsx:21]: ha... this is cool... when you have a lambda foo,
```
const foo = (a) => console.log("I'm foo, ", a);
```

and you want to use it in a then, you can do:
```
.then(resp => foo(resp));
```

but then it's the same effect as,
```
.then(foo);
```

Then takes a lambda as calls it with the response.

File: simplq/src/components/pages/Join/index.jsx (hunk 2 above)
Comment (maaverik) [reply simplq/src/components/pages/Join/index.jsx:17]: Yeah, if that's the path we're going with, then we can ditch this altogether.

File: simplq/src/components/pages/Join/index.jsx (hunk 1 above)
Comment (maaverik) [reply simplq/src/components/pages/Join/index.jsx:21]: Good to know.

Comment (thehamzarocks) [reply simplq/src/components/pages/Join/index.jsx]: This reminds me of method references in Java streams

File: simplq/src/components/pages/Status/QueueStatus.jsx [hunk 3]
```diff
@@ -13,17 +13,16 @@ import QueueDetails from './QueueDetails';
...
-function QueueStatus() {
+function QueueStatus(props) {
+  const tokenId = props.match.params.tokenId;
+  const [tokenStatusResponse, setTokenStatusResponse] = useState();
   const dispatch = useDispatch();
-  const queueName = useSelector((state) => state.appReducer.queueName);
-  const [tokenStatus, setTokenStatus] = useState();
-  const tokenId = useSelector((state) => state.appReducer.tokenId);
-  const aheadCount = useSelector((state) => state.appReducer.aheadCount);
+  dispatch(setJoinerStep(2));
```
Comment (thehamzarocks) [inline simplq/src/components/pages/Status/QueueStatus.jsx:20]: I'm wondering if instead of using magic numbers, we should define meaningful constants.

Comment (maaverik) [reply simplq/src/components/pages/Status/QueueStatus.jsx:20]: Seems like too many imports would be required in a lot of places. If that is the case, what if we just a description of what each number means in the setJoinerStep and setCreatorStep functions atleast as a temporary measure.

Comment (daltonfury42) [reply simplq/src/components/pages/Status/QueueStatus.jsx:20]: Imports are fine. We should use constants, I feel.

**Pull request merged**